dbt_subprojects/hourly_spellbook/models/_project/cow_protocol/ethereum/cow_protocol_ethereum_batches.sql

```diff
@@ -34,17 +34,19 @@ batch_counts as (
            sum(case when selector = 0x2e1a7d4d then 1 else 0 end) as unwraps,
            sum(case when selector = 0x095ea7b3 then 1 else 0 end) as token_approvals
     from {{ source('gnosis_protocol_v2_ethereum', 'GPv2Settlement_evt_Settlement') }} s
+        left outer join {{ source('cow_protocol_ethereum', 'FlashLoanRouter_evt_Settlement') }} fs
```

Comment:
This works now that we only have one alternative entry contract, but might break complexity wise if we add more entrypoints. I'm ok with it, but just want to highlight a potentially more durable way of figuring out the solver (which would be looking at traces and finding the first sender that is an allow-listed solver in the stack)

```sql
select "from"
from ethereum.traces t
join cow_protocol_ethereum.solvers s
  on t."from" = s.address
where tx_hash = 0x683fff5607fac2fd249050794fedd47875e3f94762ee885c7514b080787989c5
order by trace_address asc
limit 1
```

Reply:
Good point. That's probably safer. Although i am not fully confident with your suggestion, as it seems in the example above, we depend on the trace_address being empty and showing up first in the sorting (which i guess it is the case for now but feels risky). What i mean by that is that if you do

```
select "from"
from ethereum.traces t
join cow_protocol_ethereum.solvers s
  on t."from" = s.address
where tx_hash = 0x683fff5607fac2fd249050794fedd47875e3f94762ee885c7514b080787989c5 and cardinality(trace_address) > 0
order by trace_address asc
limit 1
```
you will get `0x9da8b48441583a2b93e2ef8213aad0ec0b392c69`

Reply:
Do you know if it's a standard thing to have the empty array always show up first in an ordering?

Reply:
> we depend on the trace_address being empty and showing up first in the sorting 

One would have to look at the default ordering semantics of arrays in SQL and if this aligns with the trace numbering (I didn't dig deep) and if not implement a more custom parsing/sorting for it, but I think the general approach should work.

From the [postgres docs](https://www.postgresql.org/docs/9.5/functions-array.html) (Presto, which is what Dune is using seems to be the same)

> The array ordering operators (<, >=, etc) compare the array contents element-by-element, using the default B-tree comparison function for the element data type, and sort based on the first difference.

Reply:
Seems this make things more explicit and works
`order by trace_address asc nulls first`

Reply:
Just pushed an update

Reply:
Here is also a [query](https://dune.com/queries/5085671), limited to the blocks around the flashloans tx that was tested on mainnet to verify that the change in this PR works as expected